spark/src/test/scala/io/ddf/spark/analytics/AggregationHandlerSuite.scala

```diff
@@ -0,0 +1,92 @@
+
+    val thrown1 = intercept[DDFException]{ddf.groupBy(List("Year").asJava, List("aaa(arrdelay)").asJava)}
+    assert(thrown1.getMessage === "undefined function aaa")
+  }
+
+  test("Proper error message for wrong Hive UDF usage") {
+    val ddf = manager.sql2ddf("select * from airline", "SparkSQL").asInstanceOf[SparkDDF]
+
+    val thrown1 = intercept[DDFException]{ddf.groupBy(List("Year").asJava, List("substring('aaaa')").asJava)}
+    assert(thrown1.getMessage === "No matching method for class org.apache.hadoop.hive.ql.udf.UDFSubstr with (string). Possible choices: _FUNC_(binary, int)  _FUNC_(binary, int, int)  _FUNC_(string, int)  _FUNC_(string, int, int)")
+
+    val thrown2 = intercept[DDFException]{ddf.groupBy(List("Year").asJava, List("to_utc_timestamp(1,1,1)").asJava)}
+    assert(thrown2.getMessage === "The function to_utc_timestamp requires two argument, got 3")
+  }
+
+  test("Proper error message for expressions or columns that contain invalid characters") {
+    val ddf = manager.sql2ddf("select * from airline", "SparkSQL").asInstanceOf[SparkDDF]
+
+    val thrown1 = intercept[DDFException]{ddf.groupBy(List("Year@").asJava, List("avg(arrdelay)").asJava)}
+    assert(thrown1.getMessage === "Expressions or columns containing invalid character @: Year@")
+
+    val thrown2 = intercept[DDFException]{ddf.groupBy(List("Year").asJava, List("avg(arrdelay@)").asJava)}
+    assert(thrown2.getMessage === "Expressions or columns containing invalid character @: avg(arrdelay@)")
+
+    val thrown3 = intercept[DDFException]{ddf.groupBy(List("Year").asJava, List("avg(arrdelay @)").asJava)}
+    assert(thrown3.getMessage === "Column or Expression with invalid syntax: 'avg(arrdelay @)'")
+
+    val thrown4 = intercept[DDFException]{ddf.groupBy(List("Year @").asJava, List("avg(arrdelay)").asJava)}
+    assert(thrown4.getMessage === "Column or Expression with invalid syntax: 'Year @'")
+
+    val thrown5 = intercept[DDFException]{ddf.aggregate("Year@,avg(arrdelay1)")}
+    assert(thrown5.getMessage === "Expressions or columns containing invalid character @: Year@")
+
+    val thrown6 = intercept[DDFException]{ddf.aggregate("Year,avg(arrdelay@)")}
+    assert(thrown6.getMessage === "Expressions or columns containing invalid character @: AVG(arrdelay@)")
+  }
+
+  test("Proper error message for new columns that contain invalid characters") {
+    val ddf = manager.sql2ddf("select * from airline", "SparkSQL").asInstanceOf[SparkDDF]
+
+    val thrown1 = intercept[DDFException]{ddf.groupBy(List("Year").asJava, List("newcol@=avg(arrdelay)").asJava)}
+    assert(thrown1.getMessage === "Expressions or columns containing invalid character @: newcol@=avg(arrdelay)")
+  }
+
+  test("Proper error message for new columns that exist") {
+    val ddf = manager.sql2ddf("select * from airline", "SparkSQL").asInstanceOf[SparkDDF]
+
+    val thrown1 = intercept[DDFException]{ddf.groupBy(List("Year").asJava, List("Year=avg(arrdelay)").asJava)}
```

Comment:
@nhanitvn @binhle the ddf.groupby return a new ddf. And in this new ddf it's entirely ok to have column named Year, arrdelay. 

Because it's new ddf. Why do we have to check if column is existed ?